javascript/node/selenium-webdriver/edge.js

```diff
@@ -65,134 +56,121 @@
   process.platform === 'win32' ? 'msedgedriver.exe' : 'msedgedriver'
 
+/** @type {remote.DriverService} */
+let defaultService = null
+
 /**
- * _Synchronously_ attempts to locate the Edge driver executable
- * on the current system. Searches for the legacy MicrosoftWebDriver by default.
- *
- * @param {string=} browserName Name of the Edge driver executable to locate.
- *   May be either 'msedge' to locate the Edge Chromium driver, or 'MicrosoftEdge' to
- *   locate the Edge Legacy driver. If omitted, will attempt to locate Edge Legacy.
- * @return {?string} the located executable, or `null`.
+ * Creates {@link selenium-webdriver/remote.DriverService} instances that manage
+ * a [ChromeDriver](https://chromedriver.chromium.org/)
+ * server in a child process.
  */
-function locateSynchronously(browserName) {
-  browserName = browserName || Browser.EDGE
-
-  if (browserName === EDGE_CHROMIUM_BROWSER_NAME) {
-    return io.findInPath(EDGEDRIVER_CHROMIUM_EXE, true)
+class ServiceBuilder extends chromium.ServiceBuilder {
+  /**
+   * @param {string=} opt_exe Path to the server executable to use. If omitted,
+   *     the builder will attempt to locate the msedgedriver on the current
+   *     PATH.
+   * @throws {Error} If provided executable does not exist, or the msedgedriver
+   *     cannot be found on the PATH.
+   */
+  constructor(opt_exe) {
+    let exe = opt_exe || locateSynchronously()
+    if (!exe) {
+      throw Error(
+        `The WebDriver for Edge could not be found on the current PATH. Please download the `+
+        `latest version of ${EDGEDRIVER_CHROMIUM_EXE} from `+
+        `https://developer.microsoft.com/en-us/microsoft-edge/tools/webdriver/ `+
+        `and ensure it can be found on your PATH.`
+      )
+    }
+    super(exe)
+    this.setLoopback(true)
   }
-
-  return process.platform === 'win32'
-    ? io.findInPath(EDGEDRIVER_LEGACY_EXE, true)
-    : null
 }
 
 /**
- * Class for managing Edge specific options.
+ * Class for managing edge chromium specific options.
  */
 class Options extends chromium.Options {
   /**
-   * Instruct the EdgeDriver to use Edge Chromium if true.
-   * Otherwise, use Edge Legacy (EdgeHTML). Defaults to using Edge Legacy.
+   * Sets the path to the edge binary to use
+   *
+   * The binary path be absolute or relative to the msedgedriver server
+   * executable, but it must exist on the machine that will launch edge chromium.
    *
-   * @param {boolean} useEdgeChromium
+   * @param {string} path The path to the edgedriver binary to use.
    * @return {!Options} A self reference.
    */
-  setEdgeChromium(useEdgeChromium) {
-    this.set(Options.USE_EDGE_CHROMIUM, !!useEdgeChromium)
-    return this
+  setEdgeChromiumBinaryPath(path) {
+    return this.setBinaryPath(path)
   }
 }
 
-Options.USE_EDGE_CHROMIUM = 'ms:edgeChromium'
-Options.prototype.BROWSER_NAME_VALUE = Browser.EDGE
-Options.prototype.CAPABILITY_KEY = 'ms:edgeOptions'
-Options.prototype.VENDOR_CAPABILITY_PREFIX = 'ms'
```

Comment:
I think this is used by the bidi tests to know where to find the websocket address for us to connect to.

Reply:
Reverted it.